EssentialApp/EssentialAppTests/ImageCommentsUIIntegrationTests.swift

```diff
@@ -0,0 +1,149 @@
+		XCTAssertTrue(sut.isShowingLoadingIndicator, "Expected loading indicator once user initiates a reload")
+
+		loader.completeImageCommentsLoadingWithError(at: 1)
+		XCTAssertFalse(sut.isShowingLoadingIndicator, "Expected no loading indicator once user initiated loading completes with error")
+	}
+
+	func test_loadImageCommentsCompletion_rendersSuccessfullyLoadedImageComments() {
+		let imageComment0 = makeImageComment(message: "message 1", createdAt: Date(), author: "author 1")
+		let imageComment1 = makeImageComment(message: "message 2 very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very long comment", createdAt: Date(), author: "author author 2")
+		let imageComment2 = makeImageComment(message: "another short message", createdAt: Date(), author: "author author author 3")
+		let imageComment3 = makeImageComment(message: "another very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very very long comment", createdAt: Date(), author: "author author author author 4")
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+		assertThat(sut, isRendering: [])
+
+		loader.completeImageCommentsLoading(with: [imageComment0], at: 0)
+		assertThat(sut, isRendering: [imageComment0])
+
+		sut.simulateUserInitiatedReload()
+		loader.completeImageCommentsLoading(with: [imageComment0, imageComment1, imageComment2, imageComment3], at: 1)
+		assertThat(sut, isRendering: [imageComment0, imageComment1, imageComment2, imageComment3])
+	}
+
+	func test_loadImageCommentsCompletion_rendersSuccessfullyLoadedEmptyImageCommentsAfterNonEmptyImageComments() {
+		let imageComment0 = makeImageComment(message: "message one", createdAt: Date(), author: "author one")
+		let imageComment1 = makeImageComment(message: "message two", createdAt: Date(), author: "author two")
+		let (sut, loader) = makeSUT()
+
+		sut.loadViewIfNeeded()
+		loader.completeImageCommentsLoading(with: [imageComment0, imageComment1], at: 0)
+		assertThat(sut, isRendering: [imageComment0, imageComment1])
+
+		sut.simulateUserInitiatedReload()
+		loader.completeImageCommentsLoading(with: [], at: 1)
+		assertThat(sut, isRendering: [])
+	}
+
+	func test_loadImageCommentsCompletion_doesNotAlterCurrentRenderingStateOnError() {
+		let imageComment0 = makeImageComment(message: "message one", createdAt: Date(), author: "author one")
```

Comment:
Consider having default values in the `makeImageComment` helper, to make tests shorter:
```suggestion
		let imageComment0 = makeImageComment()
```